plugins/lyricsviewer/__init__.py

```diff
@@ -248,8 +248,13 @@ def do_update(refresh):
                 self.set_top_box_widgets(False, True)
             return False
 
-        if self._lyrics_id != 0:
+        g_thread = GLib.main_context_get_thread_default() or \
+                        GLib.main_context_default()
+
+        if ( self._lyrics_id != 0 and 
+                g_thread.find_source_by_id(self._lyrics_id) ):
```

Comment:
This seems suspect, instead at the end of `do_update` try setting self._lyrics_id to 0.